common/client/node_fsm.go

```diff
@@ -0,0 +1,266 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+var (
+	promPoolRPCNodeTransitionsToAlive = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_alive",
+		Help: transitionString(nodeStateAlive),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToInSync = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_in_sync",
+		Help: fmt.Sprintf("%s to %s", transitionString(nodeStateOutOfSync), nodeStateAlive),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToOutOfSync = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_out_of_sync",
+		Help: transitionString(nodeStateOutOfSync),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToUnreachable = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_unreachable",
+		Help: transitionString(nodeStateUnreachable),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToInvalidChainID = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_invalid_chain_id",
+		Help: transitionString(nodeStateInvalidChainID),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToUnusable = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_unusable",
+		Help: transitionString(nodeStateUnusable),
+	}, []string{"chainID", "nodeName"})
+)
+
+// nodeState represents the current state of the node
+// Node is a FSM (finite state machine)
+type nodeState int
```

Comment:
You mean return `NodeState` instead of `string`? I'm not opposed to that, but as I said above let's make improvements of existing codebase on a separate PR.